### PR TITLE
add cloud-init path config, allow base64 encoded user-data

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -99,12 +102,13 @@ type (
...
-			Datacenter string
-			Image      string
-			UserData   string `split_words:"true"`
-			SSHKey     int
-			Token      string
-			Type       string
+			Datacenter   string
+			Image        string
+			UserData     string `split_words:"true"`
+			UserDataPath string `env:"DRONE_HETZNERCLOUD_USERDATA_PATH"`
```
Comment (bradrydzewski) [inline config/config.go]: `DRONE_HETZNERCLOUD_USERDATA_FILE` to match `DRONE_HETZNERCLOUD_USERDATA`

File: config/config.go [hunk 2]
```diff
@@ -73,20 +74,22 @@ type (
...
+			Size         string
+			Tags         []string
 		}
 
 		Google struct {
 			Zone         string `default:"us-central1-a"`
 			MachineType  string `split_words:"true" default:"n1-standard-1"`
 			MachineImage string `split_words:"true" default:"ubuntu-1510-wily-v20151114"`
 			UserData     string `split_words:"true"`
+			UserDataPath string `env:"DRONE_GOOGLE_USERDATA_PATH"`
```
Comment (bradrydzewski) [inline config/config.go]: `DRONE_GOOGLE_USERDATA_FILE`

File: config/config.go [hunk 3]
```diff
@@ -73,20 +74,22 @@ type (
...
-			UserData string `split_words:"true"`
-			Region   string
-			SSHKey   string
-			Size     string
-			Tags     []string
+			Token        string
+			Image        string
+			UserData     string `split_words:"true"`
+			UserDataPath string `env:"DRONE_DIGITALOCEAN_USERDATA_PATH"`
+			Region       string
```
Comment (bradrydzewski) [inline config/config.go]: `DRONE_DIGITALOCEAN_USERDATA_FILE`

File: config/config.go [hunk 4]
```diff
@@ -73,20 +74,22 @@ type (
...
-			Image    string
-			UserData string `split_words:"true"`
-			Region   string
-			SSHKey   string
-			Size     string
-			Tags     []string
+			Token        string
+			Image        string
+			UserData     string `split_words:"true"`
+			UserDataPath string `env:"DRONE_DIGITALOCEAN_USERDATA_PATH"`
```
Comment (bradrydzewski) [inline config/config.go]: I think we might need to remove `split_works` and specify the full environment variable name.

Comment (tboerger) [reply config/config.go]: I have not defined split words for this variable :(

**Comment by bradrydzewski** [comment]
for loading userdata from file, lets use an `_FILE` variable like what we are doing with amazon (I just added an example). In terms of base64 the existing code is good to go.

**Comment by tboerger** [comment]
I think all your comments should be applied now.

**Pull request merged**